Source/WebKit/UIProcess/WebPermissionControllerProxy.cpp

```diff
@@ -57,4 +71,42 @@ void WebPermissionControllerProxy::query(const WebCore::ClientOrigin& clientOrig
     webPageProxy->queryPermission(clientOrigin, descriptor, WTFMove(completionHandler));
 }
 
+WebPageProxy* WebPermissionControllerProxy::mostReasonableWebPageProxy(const WebCore::SecurityOriginData& topOrigin, WebCore::PermissionQuerySource source)
+{
+    WebPageProxy* webPageProxy = nullptr;
+
+    auto findWebPageProxy = [&topOrigin, &webPageProxy] (auto* processes) {
+        if (!processes)
+            return; 
+
+        for (auto& process : *processes) {
+            for (auto* potentialWebPageProxy : getPtr(process)->pages()) {
+                if (WebCore::SecurityOriginData::fromURL(URL { potentialWebPageProxy->currentURL() }) != topOrigin)
```

Comment:
You shouldn't need the `URL { }`

Reply:
I think we do need it. `currentURL()` returns a string and `fromURL()` expects a URL. Do you know if there is some way to cast between those types?